cmd/mimirtool/main.go

```diff
@@ -45,6 +46,7 @@ func main() {
 	pushGateway.Register(app, envVars)
 	remoteReadCommand.Register(app, envVars)
 	ruleCommand.Register(app, envVars)
+	backfillCommand.Register(app, envVars)
```

Comment:
I would move all the mimirtool work in a separate PR. The Mimir API work will be already huge. I think it's better to separate client and server implementation into different PRs. You can just create a draft PR with the copy of mimirtool changes from this PR. I will review it later, once Mimir API are stable, but I guess it will be useful for you in the meanwhile to manually test the API.

Reply:
That makes sense to me @pracucci - having API and client changes together is useful while prototyping.

Reply:
Done